pkg/network/render.go

```diff
@@ -338,13 +339,15 @@ func isNetworkChangeSafe(prev, next *operv1.NetworkSpec, infraRes *bootstrap.Inf
 	case len(prev.ServiceNetwork) > len(next.ServiceNetwork):
 		// Going from dual to single
 		dualStack, singleStack = prev, next
+	case !reflect.DeepEqual(prev.ServiceNetwork, next.ServiceNetwork):
+		return errors.Errorf("unsupported change to ServiceNetwork")
+	case isClusterNetworkChangeSafe(prev, next):
```

Comment:
I don't think `isClusterNetworkChangeSafe` should be a part of this switch statement, can we move that before checking `ServiceNetwork` changes and only return if it fails?
Also, please update the comments:
```
// Currently the only change we allow is switching to/from dual-stack.
// validateIPPools() will have ensured that each config is independently either
// a valid single-stack config or a valid dual-stack config. Make sure we have
// one of each.
```